test/viewmodels/journeys/adjustments/NetBusinessProfitOrLossSummarySpec.scala

```diff
@@ -17,93 +17,115 @@
-      val table = NetBusinessProfitOrLossSummary.buildAdditionsTable(profitOrLoss)(messages)
+  }
 
-      assertWithClue(result = table, expectedResult = expectedAdjustmentsTable)
+  "buildAdditionsTable must create a Table with the correct content" - {
+    forAll(additionsScenarios) { case (profitOrLoss, balancingCharge, goodsAndServices, disallowableExpenses) =>
+      s"when a net $profitOrLoss" in {
+        val table =
+          NetBusinessProfitOrLossSummary.buildAdditionsTable(profitOrLoss, balancingCharge, goodsAndServices, disallowableExpenses)(messages)
+        val expectedTable = expectedAdditionsTable(profitOrLoss, balancingCharge, goodsAndServices, disallowableExpenses)
+
+        assertWithClue(result = table, expectedResult = expectedTable)
+      }
     }
-    "buildDeductionsTable must create a Table with the correct profit or loss specific content" in {
-      val table = NetBusinessProfitOrLossSummary.buildDeductionsTable(profitOrLoss)(messages)
+  }
+
+  "buildDeductionsTable must create a Table with the correct content" - {
+    forAll(deductionsScenarios) { case (profitOrLoss, capitalAllowances, turnoverNotTaxable) =>
+      s"when a net $profitOrLoss" in {
+        val table =
+          NetBusinessProfitOrLossSummary.buildDeductionsTable(profitOrLoss, capitalAllowances, turnoverNotTaxable)(messages)
+        val expectedTable = expectedDeductionsTable(profitOrLoss, capitalAllowances, turnoverNotTaxable)
 
-      assertWithClue(result = table, expectedResult = expectedDeductionsTable)
+        assertWithClue(result = table, expectedResult = expectedTable)
+      }
     }
   }
 }
 
 object NetBusinessProfitOrLossSummarySpec {
 
-  private val defaultAmount: BigDecimal = 200
-
-  def expectedNetProfitTable()(implicit messages: Messages): Table =
-    buildTable(
-      headRow = None,
-      rows = Seq(
-        buildTableAmountRow("profitOfLoss.turnover", defaultAmount),
-        buildTableAmountRow("incomeNotCountedAsTurnover.title", defaultAmount),
-        buildTableAmountRow("profitOfLoss.totalExpenses", defaultAmount),
-        buildTableAmountRow("profitOfLoss.netProfitOrLoss.profit", BigDecimal(600))
-      ),
-      caption = Some(messages("profitOfLoss.netProfitOrLoss.profit")),
-      "govuk-!-margin-top-6 govuk-!-margin-bottom-9"
-    )
-
-  def expectedAdditionsTable()(implicit messages: Messages): Table =
-    buildTable(
-      headRow = None,
-      rows = Seq(
-        buildTableAmountRow("selectCapitalAllowances.balancingCharge", defaultAmount),
-        buildTableAmountRow("goodsAndServicesForYourOwnUse.title.individual", defaultAmount),
-        buildTableAmountRow("profitOfLoss.disallowableExpenses", defaultAmount),
-        buildTableAmountRow("profitOfLoss.totalAdditions.profit", BigDecimal(600))
-      ),
-      caption = Some(messages("profitOfLoss.additions.profit")),
-      "govuk-!-margin-bottom-9"
-    )
-
-  def expectedDeductionsTable()(implicit messages: Messages): Table =
```

Comment:
Realised that this wasn't really testing anything since the actualTable and expectedTable were just using the same methods, so switched to checking against a string output